tests/includes/juju.sh

```diff
@@ -268,3 +280,22 @@ introspect_controller() {
     echo "${idents}" | xargs -I % juju ssh -m "${name}:controller" % bash -lc "juju_engine_report" > "${TEST_DIR}/${name}-juju_engine_reports.txt"
     echo "${idents}" | xargs -I % juju ssh -m "${name}:controller" % bash -lc "juju_goroutines" > "${TEST_DIR}/${name}-juju_goroutines.txt"
 }
+
+remove_controller_offers() {
+    local name
+
+    name=${1}
+
+    OUT=$(juju models -c "${name}" --format=json | jq -r ".[\"models\"] | .[] | select(.[\"is-controller\"] == false) | .name" || true)
+    if [ -n "${OUT}" ]; then
+        echo "${OUT}" | while read -r model; do
+            OUT=$(juju offers -m "${name}:${model}" --format=json | jq -r ".[] | .[\"offer-url\"]" || true)
+            echo "${OUT}" | while read -r offer; do
+                if [ -n "${offer}" ]; then
+                    juju remove-offer -c "${name}" "${offer}"
```

Comment:
I recall having to use `--force` here if there were established relations consuming the offer.

Reply:
Ah, that's good info... I'll add it.